src/Timer_Obscure_Behaviour.md

```diff
@@ -32,99 +32,63 @@ On **CGB**:
-When TIMA overflows, the value from TMA is loaded and IF timer flag is
-set to 1, but this doesn't happen immediately. Timer interrupt is
-delayed 1 M-cycle from the TIMA overflow. The TMA reload to
-TIMA is also delayed. For 1 M-cycle, after overflowing TIMA, the value
-in TIMA is $00, not TMA. This happens only when an overflow happens, not
-when the upper bit goes from 1 to 0, it can't be done manually writing
-to TIMA, the timer has to increment itself.
-
-For example (SYS here is the lower 8 bits of the system counter):
-
-    Timer overflows:
-
-                  [A] [B]
-    SYS  FD FE FF |00| 01 02 03
-    TIMA FF FF FF |00| 23 23 23
-    TMA  23 23 23 |23| 23 23 23
-    IF   E0 E0 E0 |E0| E4 E4 E4
-
-    Timer doesn't overflow:
-
-                  [C]
-    SYS  FD FE FF 00 01 02 03
-    TIMA 45 45 45 46 46 46 46
-    TMA  23 23 23 23 23 23 23
-    IF   E0 E0 E0 E0 E0 E0 E0
-
-- During the strange cycle \[A\] you can prevent the IF flag from being
-set and prevent the TIMA from being reloaded from TMA by writing a value
-to TIMA. That new value will be the one that stays in the TIMA register
-after the instruction. Writing to DIV, TAC or other registers won't
-prevent the IF flag from being set or TIMA from being reloaded.
-
-- If you write to TIMA during the M-cycle that TMA is being loaded to it
-\[B\], the write will be ignored and TMA value will be written to TIMA
-instead.
-
-- If TMA is written the same M-cycle it is loaded to TIMA \[B\], TIMA is
-also loaded with that value.
-
-- This is a guessed schematic to explain the priorities with registers
-TIMA and TMA:
-
-![](imgs/timer_tima_tma_detailed.svg "imgs/timer_tima_tma_detailed.svg")
-
-TMA is a latch. As soon as it is written, the output shows that value.
-That explains that when TMA is written and TIMA is being incremented,
-the value written to TMA is also written to TIMA. It doesn't affect the
-IF flag though.
+Notice how the bits themselves are connected to the multiplexer and then to the falling-edge detector; this causes a few odd behaviors:
+
+- Resetting the entire system counter (by writing to `DIV`) can reset the bit currently selected by the multiplexer, thus sending a "Timer tick" and/or "[DIV-APU event](<#DIV-APU>)" pulse early.
+- Changing which bit of the system counter is selected (by changing the "Clock select" bits of [`TAC`]) from a bit currently set to another currently reset, will send a "Timer tick" pulse.
+  (For example: if the system counter is equal to \$3FF0 and `TAC` to \$FC, writing \$05 or \$06 to `TAC` will instantly send a "Timer tick", but \$04 or \$07 won't.)
+- On monochrome consoles, disabling the timer if the currently selected bit is set, will send a "Timer tick" once.
+  This does not happen on Color models.
+- On Color models, a write to `TAC` that fulfills the previous bullet's conditions *and* turns the timer on (it was disabled before) may or may not send a "Timer tick".
+  This depends on a race condition, so it is device-dependent.
```

Comment:
I'm not sure what this line is adding, since it's a "race condition" for which no additional details are provided (device dependent in what way?)

Reply:
What I want to convey here is that from device to device, the behaviour may be consistent one way or another, *or be inconsistent entirely*. This is what the "race condition" statement is intended to convey.

Reply:
I would say inconsistent better conveys the message, then. "Since it depends on a race condition it's device-dependent" is a very confusing phrasing to me, as I don't see why a race condition should imply that is device-dependent (race condition should be used as a synonim to "indeterministic behaviour due to timing issues or other _incontrollable_ variables)

Reply:
If the fact that it's device-dependent really depends on the fact that it's a race condition, then I would expand and explain why, if relevant. Otherwise, back to inconsistency as a term.

Reply:
It is, internally, a race condition between some signals; but I guess technical correctness/precision here is more confusing than papering over details, yeah.